### PR TITLE
Dynamic offsetting.

### PR DESCRIPTION
**Pull request by vext01**
Requires https://github.com/ykjit/ykllvm/pull/152.

The main work is the second commit.

### CODE REVIEW

File: tests/c/simple_interp_loop2.newcg.c [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+//     pc=2, mem=2
+//     pc=3, mem=1
+//     pc=0, mem=1
+//     pc=1, mem=1
+//     pc=2, mem=1
+//     pc=3, mem=0
+//     jitstate: deoptimise
+//     pc=4, mem=0
+//     pc=5, mem=0
+
```
Comment (ptersilie) [inline tests/c/simple_interp_loop2.newcg.c:28]: Does this need a description what this test does (since we have one for all other tests)?

**Comment by ptersilie** [comment]
Looks good. Just one small comment.

File: tests/c/simple_interp_loop2.newcg.c (hunk 1 above)
Comment (vext01) [reply tests/c/simple_interp_loop2.newcg.c:28]: The lack of description was inherited from the old codegen test. I added commits to fix both.

**Comment by vext01** [comment]
OK to squash?

**Comment by ptersilie** [comment]
Please squash.

**Comment by ptersilie** [comment]
Please squash.

**Comment by vext01** [comment]
splat.

**Pull request merged**